src/main/java/io/camunda/zeebe/bpmnassert/testengine/RecordStreamSourceImpl.java

```diff
@@ -24,26 +30,31 @@
 
 public class RecordStreamSourceImpl implements RecordStreamSource {
 
-  private final Iterable<Record<?>> records;
+  private final LogStreamReader logStreamReader;
+  private final int partitionId;
+  private final List<Record<?>> records = new ArrayList<>();
+  private long lastPosition = -1L;
 
-  public RecordStreamSourceImpl(final Iterable<Record<?>> records) {
-    this.records = records;
+  public RecordStreamSourceImpl(final LogStreamReader logStreamReader, final int partitionId) {
+    this.logStreamReader = logStreamReader;
+    this.partitionId = partitionId;
+  }
+
+  @Override
+  public Iterable<Record<?>> records() {
+    updateWithNewRecords();
+    return records;
```

Comment:
:x: Please return `Collections.unmodifiableList(records)` or something like this. Now that the records get updated, we need to protect the list against modification from the outside.